File: NewsApp/app/src/main/java/com/example/android/topic41/data/network/Network.java
package com.example.android.topic41.data.network;

import androidx.annotation.NonNull;

import com.example.android.topic41.domain.util.Article;
import com.example.android.topic41.domain.util.News;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import io.reactivex.Single;

public class Network implements NetworkInterface {
    private JsonPlaceHolderApi jsonPlaceHolderApi;
    private Single<List<Article>> observable;
    private Map<String, String> parameters = new HashMap<>();

    private final String KEY = "apiKey";
    private final String KEY_VALUE = "eb3c938655f74099948faa2b250623df";



    public Network(JsonPlaceHolderApi jsonPlaceHolderApi) {
        this.jsonPlaceHolderApi = jsonPlaceHolderApi;
    }

    @Override
    public Single<List<Article>> createObservable(@NonNull String theme) {
        parameters.put(KEY, KEY_VALUE);
        parameters.put("q", theme);
        observable = jsonPlaceHolderApi.getNews(parameters).map(News::getArticles);
        return observable;
    }
}

File: NewsApp/app/src/main/java/com/example/android/topic41/data/network/Network.kt
package com.example.android.topic41.data.network

import com.example.android.topic41.domain.util.Article
import com.example.android.topic41.domain.util.News
import io.reactivex.Single
import java.util.*


class Network(private val jsonPlaceHolderApi: JsonPlaceHolderApi) : NetworkInterface {
    private lateinit var observable: Single<List<Article>>
    private val parameters: MutableMap<String, String> = HashMap()


    override fun createObservable(theme: String): Single<List<Article>> {
        parameters[KEY] = KEY_VALUE
        parameters["q"] = theme
        observable = jsonPlaceHolderApi.getNews(parameters).map(News::articles)
        return observable
    }

    companion object {
        private const val KEY = "apiKey"
        private const val KEY_VALUE = "eb3c938655f74099948faa2b250623df"
    }
}
